### PR TITLE
Remove Avere and take NFS servers 

### PR DESCRIPTION
**Pull request by lawrencegripper**
PR for #1207 

Currently work in progress. 

Based off #1184 to avoid merge issues. 

Todo: 
- [x] Remove Avere Template 
- [x] Move existing Avere script references
- [x] Add NFS server param
- [x] Update mount script to handle NFS mount options

### CODE REVIEW

**Comment by lawrencegripper** [comment]
Quick update. I've tested this with `AzureFiles` and seems to behave as expected. Just running now with an NFS server to test that out.

**Comment by marrobi** [comment]
Can you do #1021 as part of this? Thanks.

**Comment by lawrencegripper** [comment]
@marrobi  I'll pick that up as a separate PR probably.

**Comment by lawrencegripper** [comment]
@marrobi Good news - I tested this out with an HA NFS deployment from here https://azure.microsoft.com/en-us/resources/templates/nfs-ha-cluster-ubuntu/ and everything worked fine. 

While debugging the deployment I tried to write up the steps I was taking so they're captured in the README for others with some additional guidance around common failure issues. 

@martinpeck Would you be alright to have a quick look at the documentation changes in here and let me know where my english has failed :pray:

File: opencga-app/app/scripts/azure/arm/README.md [hunk 1]
```diff
@@ -19,6 +19,177 @@ Click the following link the ensure you fill in the parameters according to thei
...
+This resource group contains an HA SOLR cluster used by OpenCGA for indexing the data in `HBASE`. To provide the leadership election and failover a small `zookeeper` HA deployment is also present in this resource group. 
+
+## Debugging deployment Issues
+
+
+Some failure cases will result in the CLI returning a failed deployment but others aren't detectable from the CLI in the current version. The best way to debug these further is to either connect to the `daemonvm` or review the `log analytics` data. 
+
+### Log Analytics Checks
+
+Under the main resource group for the deployment several `solutions` are deployed for `loganalytics`. One of particualar use [is the Container Monitoring Solution])https://docs.microsoft.com/en-us/azure/azure-monitor/insights/containers) which gives an overview of the logs, usage and status of the containers deployed in the solution.
```
Comment (bart-jansen) [inline opencga-app/app/scripts/azure/arm/README.md]: typo `particualar` should be `particular`

File: opencga-app/app/scripts/azure/arm/README.md [hunk 2]
```diff
@@ -19,6 +19,177 @@ Click the following link the ensure you fill in the parameters according to thei
...
+
+### Debugging Azure Batch / Transform Load failures
+
+Having submitted a job such as:
+
+`sudo /opt/opencga/bin/opencga.sh variant index --file ALL.chr22.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --calculate-stats --annotate --index-search -o outDir`
+
+You can look at Azure Batch to observe the task execution and review the log outputs. The recommended way to do this [is to install Azure Batch Explorer](https://azure.github.io/BatchExplorer/). 
+
+Once installed and logged in navigate to the `Jobs` and select the current job then under `Tasks` you will see each execution and can drill down to retrieve the logs and follow it's execution.
```
Comment (bart-jansen) [inline opencga-app/app/scripts/azure/arm/README.md]: `it's execution` -> `its execution`. Maybe also add a comma after `current job`

File: opencga-app/app/scripts/azure/arm/README.md [hunk 3]
```diff
@@ -19,6 +19,177 @@ Click the following link the ensure you fill in the parameters according to thei
...
+
+In this example we see a failure occur around `10:20UTC` and the various web server nodes report their failures. 
+
+![healthcheck graph](docs/healthcheck.jpg)
+
+
+### Daemon VM checks and FAQ
+
+Q. How do I connect to the web servers hosting the `tomcat` app. 
+A. You can use the `daemonvm` as a jumpbox. Retreive the `daemonvm`'s public IP from the portal and use `ssh -J opencgaadmin@DAEMONVM_PUBLIC_IP_HERE opencgaadmin@opencga000001` to connect to the instances. You can change increment `opencga000001` to connect to other instances of the web servers.
```
Comment (bart-jansen) [inline opencga-app/app/scripts/azure/arm/README.md]: `Retreive` -> `Retrieve`

File: opencga-app/app/scripts/azure/arm/README.md [hunk 4]
```diff
@@ -19,6 +19,177 @@ Click the following link the ensure you fill in the parameters according to thei
...
+A. `cloud-init status` should return a status of `done` and `sudo docker ps -a` should return 
+
+```
+CONTAINER ID        IMAGE                                 COMMAND                  CREATED             STATUS                             PORTS                                                    NAMES
+e948dabbc0ec        lawrencegripper/opencga-daemon:nfs1   "/bin/bash -c 'echo …"   19 minutes ago      Up 57 seconds (health: starting)                                                            opencga-daemon
+a37fce3165dd        lawrencegripper/opencga-init:nfs1     "/bin/bash /tmp/setu…"   20 minutes ago      Exited (0) 19 minutes ago                                                                   opencga-init
+2d00c901c3e9        microsoft/oms                         "/opt/main.sh"           23 minutes ago      Up 22 minutes                      127.0.0.1:25224->25224/udp, 127.0.0.1:25225->25225/tcp   omsagent
+
+```
+If the `opencga-init` container has an exit code other than `0` of the `opencga-daemon` `status` is `Restarting` then use `sudo docker logs opencga-init` or `sudo docker logs opencga-daemon` to review the logs.
```
Comment (bart-jansen) [inline opencga-app/app/scripts/azure/arm/README.md]: typo: `of the opencga-daemon status` should probably be `or the opencga-daemon status`

File: opencga-app/app/scripts/azure/arm/README.md [hunk 5]
```diff
@@ -124,11 +295,16 @@ Additionally you can deploy a custom size by specifying the `customDeploymentSiz
         }
 ```
 
-## Additional Notes
+## Developing 
+
+In many of the templates scripts are embedded in the ARM as `base64` `gzip`'d strings. 
 
-### Avere - First Run
+When making changes to the scripts present on disk the [following steps need to be taken](mongodb/README.MD) to ensure the update script is set in the ARM.
```
Comment (bart-jansen) [inline opencga-app/app/scripts/azure/arm/README.md]: `in the ARM` -> `in the ARM template`

File: opencga-app/app/scripts/azure/arm/README.md [hunk 6]
```diff
@@ -124,11 +295,16 @@ Additionally you can deploy a custom size by specifying the `customDeploymentSiz
...
-## Additional Notes
+## Developing 
+
+In many of the templates scripts are embedded in the ARM as `base64` `gzip`'d strings. 
 
-### Avere - First Run
+When making changes to the scripts present on disk the [following steps need to be taken](mongodb/README.MD) to ensure the update script is set in the ARM. 
 
-To run Avere you must first accept the legal terms of the license using the following command `az vm image accept-terms --urn microsoft-avere:vfxt:avere-vfxt-controller:latest`
+You can also use some additional tricks to get more information about such as the following command.
```
Comment (bart-jansen) [inline opencga-app/app/scripts/azure/arm/README.md]: missing word. maybe  `more information about the deployment` ?

File: opencga-app/app/scripts/azure/arm/README.md [hunk 7]
```diff
@@ -124,11 +295,16 @@ Additionally you can deploy a custom size by specifying the `customDeploymentSiz
...
 
-To run Avere you must first accept the legal terms of the license using the following command `az vm image accept-terms --urn microsoft-avere:vfxt:avere-vfxt-controller:latest`
+You can also use some additional tricks to get more information about such as the following command. 
 
-See [the Avere docs for more details](https://docs.microsoft.com/en-us/azure/avere-vfxt/avere-vfxt-prereqs#accept-software-terms-in-advance)
+```
+az deployment create --location westeurope --template-file azuredeploy.json --parameters @azuredeploy.parameters.private.json --name $RANDOM --parameters _artifactsLocationSasToken=?$RANDOM --verbose; notify-send -u critical "Deploy finished with exit code: $?"
+```
 
+The `verbose` setting will give you detailed information on failures. `$RANDOM` is used to ensure that deployment names don't clash. `notify-send` will push a notification to `ubuntu` when a deployment has either failed or succeeded so you don't miss it. `_artifactsLocationSasToken=?$RANDOM` ensures that `github` hosted scripts aren't cached incorrectly when working.
```
Comment (bart-jansen) [inline opencga-app/app/scripts/azure/arm/README.md]: `when working`... (missing words?). Maybe `when working with different versions`

File: opencga-app/app/scripts/azure/arm/azuredeploy.json [hunk 8]
```diff
@@ -209,16 +191,33 @@
...
+        },
+        "fileSystemType": {
+            "type": "string",
+            "defaultValue": "azurefiles",
+            "allowedValues": [
+                "azurefiles",
+                "nfs"
+            ],
+            "metadata": {
+                "description": "Defines the file system type the sytem will use when deployed in Azure. When using `NFS` the servers must have a mountable folder `opencga` available for mouting by the USER: `opencga` UID: `1001`"
```
Comment (bart-jansen) [inline opencga-app/app/scripts/azure/arm/azuredeploy.json]: typo: `mouting` -> `mounting`

**Comment by lawrencegripper** [comment]
@bart-jansen Thanks so much for this review it was ace and sorry I should have caught more of these ones myself. 

@marrobi and @martinpeck you alright for me to merge this one at some point today?

**Comment by marrobi** [comment]
Yup, haven't testedit, but providing it works, get it merged! On cellbase today, so will need a new deployment later, and can try it out then.

**Comment by lawrencegripper** [comment]
Cool - I'll wait for the build to finish then merge it in. @marrobi anything you think it worth adding into the additional docs? Put down all my standard debugging techniques etc.

**Comment by marrobi** [comment]
@lawrencegripper did we discuss  that azure files was still used for configuration store and the external NFS if provided was used for variant storage? Does this PR take Azure Files out completely?

**Comment by lawrencegripper** [comment]
Sorry for the confusion I thought that splitting out the config file was a separate bit of work so that hasn't happened here. Either Azure files is auto configured or NFS is configured but both will hold the Config and data

**Pull request merged**